src/libcore/ffi.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Implementation of a `va_list`.
+
+#[cfg(any(target_arch = "arm", target_arch = "mips", target_arch = "powerpc64",
+          target_arch = "x86", windows))]
+#[cfg(not(stage0))]
+/// Basic byte pointer variant of a `va_list`.
+pub type VaListImpl = *const i8;
```

Comment:
I believe this intermediate type doesn't need to be public, right?

Reply:
Good catch